src/components/templates/Search/utils.ts

```diff
@@ -66,42 +66,72 @@ export function getSearchQuery(
       `(service.attributes.additionalInformation.categories:\"${categories}\")`
     : text || ''
 
+  if (searchTerm.startsWith('did:op:')) {
+    searchTerm = searchTerm.substring(7)
+  }
   // HACK: resolves the case sensitivity related to dataTokenInfo.symbol
   searchTerm = '*' + searchTerm.toUpperCase() + '*'
   searchTerm = addTypeFilterToQuery(searchTerm, serviceType)
-
   return {
     page: Number(page) || 1,
     offset: Number(offset) || 21,
     query: {
-      query_string: {
-        query: `${searchTerm} -isInPurgatory:true`,
-        fields: [
-          'dataTokenInfo.name',
-          'dataTokenInfo.symbol',
-          'service.attributes.main.name',
-          'service.attributes.main.author',
-          'service.attributes.additionalInformation.description'
-        ],
-        default_operator: 'AND'
+      bool: {
+        must: [
+          {
+            bool: {
+              should: [
+                {
+                  query_string: {
+                    query: `${searchTerm}`,
+                    fields: [
+                      'id',
+                      'publicKey.owner',
+                      'dataToken',
+                      'dataTokenInfo.name',
+                      'dataTokenInfo.symbol',
+                      'service.attributes.main.name^10',
+                      'service.attributes.main.author',
+                      'service.attributes.additionalInformation.description',
+                      'service.attributes.additionalInformation.tags'
+                    ],
+                    default_operator: 'AND',
```

Comment:
remove this line, it shouldn't be `AND`

Reply:
I added 'AND' as default op because using OR would give a match for every word in search string. For "worldlist used by" I got matches for assets that only contain "used" in their description/ title/ etc

Reply:
yes, please used the searchTerm formated as described above with 'OR' between each word. "worldlist used by" would become "worldlist OR used OR by", just in `query_string`, not in `match_phrase`

Reply:
Didn't used 2 different search terms on `query_string` and `match_phrase ` there's where the problem came from. Please ignore the comments.